### ISSUE TITLE
Cannot Select the UnityJoystickManager Input Data Provider

### ISSUE THREAD

**Issue by wlewis55** [microsoft/MixedRealityToolkit-Unity#9931]
I am not able to select the _MixedReality.Toolkit.UnityInput.UnityJoystickManager_ Provider. Below is a picture of the provider I am trying to select.

![image](https://user-images.githubusercontent.com/4706315/120539446-97e81b80-c3b5-11eb-93ab-53ef1809030a.png)

## To reproduce

1. Create a new MRTK project.
2. Create a new MR Toolkit config profile, and clone a MRInputSystemProfile to be able to customize the input values.
3. Under Input Data Providers, create a new data provider.
4. For Type, Select the UnityJoystickManager.

## Expected behavior

The Type should populate with the selected provider. I need this provider to be able to connect a Xbox controller to the Hololens through MRTK.

## Screenshots
Here is a video of the Provider deleing it's instance after being selected. (I was not able to record the actual menu but it is the same as show above). First, I add the UnityTouchDeviceManager to show the Type field will populate, but then the instance is delete when selecting the UnityJoystickManager.

![unityjoystickmanager](https://user-images.githubusercontent.com/4706315/120539796-f3b2a480-c3b5-11eb-8dab-17b87c1ad5b3.gif)

## Setup

- **Unity 2020.3.11f1** (I did not have this problem in Unity 2019)
- MRTK Version 2.7.0

## Target platform

- Unity Editor
- HoloLens 2
- Xbox Controller

## Additional context

There is very little documentation on this provider and connecting Input Controllers.

**Comment by wlewis55** [comment]
After doing some testing I believe this is only a problem with 2.7.0

**Comment by tjhappy** [comment]
My test found that the latest version of hololens2 does not support Xbox one, but the old version can use .

**Comment by RogPodge** [comment]
Figured out why this occurs, the Joystick Manager is classified as a Legacy XR device manager, so adding it while the XR SDK tab is selected does not show that it is added.

![DeviceManagerBug](https://user-images.githubusercontent.com/39840334/121102437-6b1b7600-c7b2-11eb-9be5-666f66b21a0f.gif)


I'm not entirely sure why it is locked to Legacy XR, @keveleigh  any insights on this?

**Comment by wlewis55** [comment]
@RogPodge I had a feeling that's what is happening... but I didn't have the Legacy tab (see the first image above). Do you know if I have to have OpenXR installed to get that? My current projects are setup with the built in Unity plugins.

**Comment by RogPodge** [comment]
Oh, since Unity 2020 no longer supports Legacy XR, the tab won't show up. All the more reason for us to double check that the joystick manager works on the XR SDK pipeline

**Comment by keveleigh** [comment]
> the Joystick Manager is classified as a Legacy XR device manager, so adding it while the XR SDK tab is selected does not show that it is added.

Yep, this is exactly right! It looks like I may have prematurely locked it to legacy though, as it's technically just dependent on Unity's old input system. We have #9703 to track updating MRTK to the new input system, which is Unity's default for things like OpenXR going forward, but it _is_ still possible to run them side-by-side (at a perf cost, I assume).

@RogPodge I think it's reasonable to unmark this provider as legacy-only, at least until we update MRTK to the new input system (and depending on how that's done, whether side-by-side using `#if`s or as a separate provider, maybe indefinitely!)

**Comment by RogPodge** [comment]
I'll unmark it today!
